xonsh/environ.py

```diff
@@ -2267,6 +2276,8 @@ def xonsh_script_run_control(filename, ctx, env, execer=None, login=True):
         msg = "error running xonsh run control file {0!r}: {1!s}"
         print_exception(msg.format(filename, err))
         loaded = False
+    finally:
+        sys.path = list(filter(lambda p: not isinstance(p, _RcPath), sys.path))
```

Comment:
Also how about checking for identity like so:
```suggestion
        sys.path = list(filter(lambda p: p is not rc_dir, sys.path))
```
This way if somehow an RC file is run inside another RC file their `sys.path` objects won't interfere with each other

Reply:
Didn't think about the possibility of rc files running one another, great point! I committed your suggestion.